File: 02 - Cpp/02 - Idades.cpp
#include <iostream>
#include <iomanip>
#include <climits>

using namespace std;

int main()
{
    int idade1, idade2;
    string nome1, nome2;
    double media;

    cout << "Dados da primeira pessoa: " << '\n';
    cout << "Nome: ";
    getline(cin,nome1);
    cout << "Idade: ";
    cin >> idade1;

    cout << "Dados da segunda pessoa: " << '\n';
    cout << "Nome: ";
    cin.ignore(INT_MAX, '\n');
    getline(cin,nome2);
    cout << "Idade: ";
    cin >> idade2;

    media = (idade1 + idade2) / 2.0;

    cout << fixed << setprecision(1);
    cout << "A idade media de " << nome1 << " e "
         << nome2 << " eh de " << media << " anos" << '\n';
    return 0;
}

File: 04 - Csharp/02 - Idades.cs
using System;
using System.Globalization;

namespace Programa
{
    class program
    {
        static void Main(string[] args)
        {
            CultureInfo CI = CultureInfo.InvariantCulture;

            int idade1, idade2;
            double media;
            string nome1, nome2;

            Console.WriteLine("Dados da primeira pessoa: ");
            Console.Write("Nome: ");
            nome1 = Console.ReadLine();
            Console.Write("Idade: ");
            idade1 = int.Parse(Console.ReadLine());
            
            Console.WriteLine("Dados da segunda pessoa: ");
            Console.Write("Nome: ");
            nome2 = Console.ReadLine();
            Console.Write("Idade: ");
            idade2 = int.Parse(Console.ReadLine());

            media = (double) (idade1 + idade2) / 2;

            Console.WriteLine("A idade media de " + nome1 + " e " + nome2
                             + " eh de " + media.ToString("F1", CI) + " anos");
        }
    }
}